applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,242 +60,273 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
-        return unlockKeyValue(key, null, null);
+        return unlockKeyValue(key, null);
     }
 
     /**
-     * Unlocks ConfigurationSetting. If present, label must be explicit label value (not a wildcard).
-     * For all operations it's an optional parameter. If omitted it implies null label.
-     * @param key key name
-     * @return ConfigurationSetting
+     * Unlocks a ConfigurationSetting with a matching {@param key}, optional {@param label}. If present, {@param label}
+     * must be explicit label value (not a wildcard).
+     *
+     * @param key   key name
+     * @param label Optional. If not specified, {@link ConfigurationSetting#NULL_LABEL} is used. If specified, it must
+     *              be an explicit value and cannot contain wildcard characters.
+     * @return ConfigurationSetting that was unlocked.
      */
-    public Mono<RestResponse<ConfigurationSetting>> unlockKeyValue(String key, String label, ETagFilter filter) {
+    public Mono<RestResponse<ConfigurationSetting>> unlockKeyValue(String key, String label) {
         if (key == null || key.isEmpty()) {
             throw new IllegalArgumentException("Parameter key is required and cannot be null or empty");
+        } else if (label == null) {
+            label = ConfigurationSetting.NULL_LABEL;
         }
-        if (filter != null) {
-            return service.unlockKeyValue(baseUri.toString(), key, label, filter.ifMatch(), filter.ifNoneMatch());
-        }
+
         return service.unlockKeyValue(baseUri.toString(), key, label, null, null);
     }
 
     /**
-     * Lists chronological/historical representation of ConfigurationSetting resource(s). Revisions eventually expire (default 30 days).
-     * For all operations key is optional parameter. If ommited it implies any key.
-     * For all operations label is optional parameter. If ommited it implies any label.
+     * Lists the ConfigurationSettings.
+     *
      * @param filter query options
-     * @return Revisions of the ConfigurationSetting
+     * @return KeyValues
      */
-    public Flux<ConfigurationSetting> listKeyValueRevisions(RevisionFilter filter) {
-        return listKeyValueRevisions(filter, pageResponseFlux -> pageResponseFlux.map(r -> r.items())
-                                                              .flatMapIterable(i -> i));
+    public Flux<ConfigurationSetting> listKeyValues(KeyValueListFilter filter) {
+        Mono<RestResponse<PageImpl<ConfigurationSetting>>> result;
+        if (filter != null) {
+            result = service.listKeyValues(baseUri.toString(), filter.key(), filter.label(), filter.fields(), filter.acceptDateTime(), filter.range());
+        } else {
+            result = service.listKeyValues(baseUri.toString(), null, null, null, null, null);
+        }
+
+        return result.flatMapMany(p -> Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode())))
```

Comment:
Were you still planning to introduce a common type for RestPagedResponseImpl?

Reply:
If not, at the very least a convenience method inside this class (so, a private method) would be nice, something like

```
private <T> RestPagedResponseImpl<T> toPagedResponse(PageImpl<T> p) {
    return new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode()));
}
```

This would mean you could write `result.flatMapMany(this::toPagedResponse)`

Reply:
Yeah. I am going to add that common RestPagedResponse to the common-library.

Reply:
I created an issue to resolve this #3135